clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -152,14 +152,29 @@ public Password getPassword(String key) {
      * @return a Map containing the settings with the prefix
      */
     public Map<String, Object> originalsWithPrefix(String prefix) {
-        Map<String, Object> result = new RecordingMap<>(prefix);
+        Map<String, Object> result = new RecordingMap<>(prefix, false);
         for (Map.Entry<String, ?> entry : originals.entrySet()) {
             if (entry.getKey().startsWith(prefix) && entry.getKey().length() > prefix.length())
                 result.put(entry.getKey().substring(prefix.length()), entry.getValue());
         }
         return result;
     }
 
+    /**
+     * Put all keys that do not start with {@code prefix} and their parsed values in the result map and then
+     * put all the remaining keys with the prefix stripped and their parsed values in the result map.
+     *
+     * This is useful if one wants to allow prefixed configs to override default ones.
+     */
+    public Map<String, Object> valuesWithPrefixOverride(String prefix) {
+        Map<String, Object> result = new RecordingMap<>(values(), prefix, true);
+        for (Map.Entry<String, ?> entry : values.entrySet()) {
```

Comment:
Perhaps values need to be originals since values only include keys defined in the config def?